dimension/request.go

```diff
@@ -57,47 +55,16 @@ func (request *Request) Post(httpClient *http.Client) error {
 	}
 	req.Header.Set("internal-token", request.DatasetAPIAuthToken)
 
-	res, err := httpClient.Do(req)
+	res, err := httpClient.Do(ctx, req)
 	if err != nil {
-		if nextError := request.retryRequest(httpClient, err); nextError != nil {
-			return nextError
-		}
-
-		return nil
+		return err
 	}
-
 	defer res.Body.Close()
 
 	if res.StatusCode != http.StatusOK {
-		errorInvalidStatus := fmt.Errorf("invalid status [%d] returned from [%s]", res.StatusCode, request.DatasetAPIURL)
-
-		// If request fails due to an internal server error from
-		// Dataset API try again and increase the backoff
-		if res.StatusCode != http.StatusInternalServerError {
-			return errorInvalidStatus
-		}
-
-		if err := request.retryRequest(httpClient, errorInvalidStatus); err != nil {
-			return err
-		}
+		return fmt.Errorf("invalid status [%d] returned from [%s]", res.StatusCode, request.DatasetAPIURL)
```

Comment:
log.Data with KV pairs would be preferable to a formatted string as it is more structured

Reply:
This is not a log, it's building an error to indicate a problem with request.